Node.js APM docs updates (#9333)
DataDog/documentation: content/en/tracing/setup_overview/custom_instrumentation/nodejs.md

You may also want to extend the functionality of the `dd-trace` library or gain finer control over instrumenting your application. Several techniques are provided by the library to accomplish this.

## Adding tags

The built-in instrumentation and your own custom instrumentation will create
spans around meaningful operations.

{{< tabs >}}
{{% tab "Locally" %}}

> You can access the active span in order to include meaningful data via tags.

Reviewer: ```suggestion
You can access the active span in order to include meaningful data by adding tags.
```

```javascript
const span = tracer.scope().active()
```

API details for `Scope` can be found [here][1].

> Tags can be added to a span using the `setTag` or `addTags` method on a span. Supported value types are string, number and object.

Reviewer: ```suggestion
You can add tags to a span using the `setTag` or `addTags` method on a span. Supported value types are string, number, and object.
```

```javascript
// add a foo:bar tag
span.setTag('foo', 'bar')

// add a user_id:5 tag
span.setTag('user_id', 5)

// add a obj.first:foo and obj.second:bar tags
span.setTag('obj', { first: 'foo', second: 'bar' })

// add a foo:bar and baz:qux tags
span.addTags({
  foo: 'bar',
  baz: 'qux'
})
```

[1]: https://datadoghq.dev/dd-trace-js/interfaces/scope.html
{{% /tab %}}

{{% tab "Globally" %}}

> Tags can be added to every span by configuring them directly on the tracer. This can be done with the comma-separated `DD_TAGS` environment variable or with the `tags` option on the tracer initialization.

Reviewer: ```suggestion
You can add tags to every span by configuring them directly on the tracer, either with with the comma-separated `DD_TAGS` environment variable or with the `tags` option on the tracer initialization:
```

```javascript
// equivalent to DD_TAGS=foo:bar,baz:qux
tracer.init({
  tags: {
    foo: 'bar',
    baz: 'qux'
  }
})

// All spans will now have these tags
```

{{% /tab %}}

{{% tab "Component" %}}

Some of our integrations support span hooks that can be used to update the span right before it's finished. This is useful to modify or add tags to a span that is otherwise inaccessible from your code.

```javascript
// at the top of the entry point right after tracer.init()
tracer.use('express', {
  // hook will be executed right before the request span is finished
  hooks: {
    request: (span, req, res) => {
      span.setTag('customer.id', req.query.customer_id)
    }
  }
})
```

API details for individual plugins can be found [here][1].

[1]: https://datadoghq.dev/dd-trace-js/modules/plugins.html
{{% /tab %}}

{{% tab "Errors" %}}

> Errors can be added to a span with the special `error` tag that supports error objects. This will split the error in 3 different tags: `error.type`, `error.msg` and `error.stack`.

Reviewer: ```suggestion
Errors can be added to a span with the special `error` tag that supports error objects. This will split the error into three tags: `error.type`, `error.msg` and `error.stack`.
```
